### ISSUE TITLE
Error in ResNet ImageNet examples

### ISSUE THREAD

**Issue by samuelstevens** [mosaicml/examples#219]
While training ResNet with the "mild" ImageNet recipe, I realized that the call to `config.update(recipe_config)` doesn't actually work. [See these lines.](https://github.com/mosaicml/examples/blob/132ec02cc2c75e66a410f869b75697a6219fedae/examples/resnet_imagenet/main.py#LL43-L49C37)

`recipe_config` is an OmegaConf dictionary config with keys that include dots in the name:

```py
{
    'model.loss_name': 'binary_cross_entropy', 
    'train_dataset.crop_size': 176, 
    'eval_dataset.resize_size': 232, 
     'max_duration': '36ep'
}
```

When you call `config.update(recipe_config)`, it adds these keys to the `config` object directly, instead of updating the`config.train_dataset` nested-dictionary. This means the `max_duration` is actually 36ep because it's a top-level key, but the crop sizes will not be changed and the model loss is still cross entropy instead of the binary variant.

You can fix it like this:

```py
for key, value in recipe_config.items():
    OmegaConf.update(config, key, value)
```

`OmegaConf.update` respects the dots in the key names.

**Comment by Landanjs** [comment]
Thanks for identifying this and providing a fix! We will merge the fix in a PR soon. Apologies for the bug 😅
